test-clients/src/main/java/com/hedera/services/bdd/suites/token/UniqueTokenManagementSpecs.java

```diff
@@ -377,43 +374,43 @@ private HapiApiSpec happyPathWithRepeatedMetadata() {
                                 .initialSupply(0)
                                 .supplyKey("supplyKey")
-                                .treasury(TOKEN_TREASURY),
-                        mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo"))),
-                        tokenAssociate(FIRST_USER, A_TOKEN),
-                        tokenAssociate(SECOND_USER, A_TOKEN)
-                ).when(
-                        cryptoTransfer(
-                                movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER),
-                                movingUnique(1, A_TOKEN).between(FIRST_USER, SECOND_USER)
-                        ).signedBy("signingKeyTreasury", "signingKeyFirstUser", DEFAULT_PAYER).via("cryptoTransferTxn")
-                ).then(
-                        getAccountBalance(TOKEN_TREASURY)
-                                .hasTokenBalance(A_TOKEN, 0),
-                        getAccountBalance(FIRST_USER)
-                                .hasTokenBalance(A_TOKEN, 0),
-                        getAccountBalance(SECOND_USER)
-                                .hasTokenBalance(A_TOKEN, 1),
-
-                        getTokenNftInfo(A_TOKEN, 1)
-                                .hasSerialNum(1)
-                                .hasMetadata(ByteString.copyFromUtf8("memo"))
-                                .hasTokenID(A_TOKEN)
-                                .hasAccountID(SECOND_USER),
-                        getTxnRecord("cryptoTransferTxn").logged()
+                                .treasury(TOKEN_TREASURY)
+                )
+                .when(
+                        mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo"))).via("mintTransferTxn")
+                )
+                .then(
+                        UtilVerbs.withOpContext((spec, opLog) -> {
+                            var mintNft = getTxnRecord("mintTransferTxn");
+                            allRunFor(spec, mintNft);
+                            var tokenTransferLists = mintNft.getResponseRecord().getTokenTransferListsList();
+                            tokenTransferLists.stream().forEach(tokenTransferList -> {
+                                Assert.assertNotNull(tokenTransferList.getNftTransfersList());
+                                tokenTransferList.getNftTransfersList().stream().forEach(nftTransfers -> {
+                                    Assert.assertEquals(
+                                            "0.0.0",
+                                            String.format("%s.%s.%s",
+                                                    nftTransfers.getSenderAccountID().getAccountNum(),
+                                                    nftTransfers.getSenderAccountID().getRealmNum(),
+                                                    nftTransfers.getSenderAccountID().getShardNum()));
+                                    Assert.assertNotEquals(0L, nftTransfers.getReceiverAccountID().getAccountNum());
```

Comment:
I think that you can assert the exact number for the treasury by:
```
Assert.assertEquals(TxnUtils.asId(TOKEN_TREASURY, spec), nftTransfers.getReceiverAccountID());
```

Reply:
Done